### PR TITLE
[incubator/elasticsearch] adds elasticsearch chart

### PR DESCRIPTION
**Pull request by sameersbn**
The chart deploys a single node elasticsearch deployment

### CODE REVIEW

File: incubator/elasticsearch/templates/NOTES.txt [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{{- if contains .Values.serviceType "LoadBalancer" }}
+
+-------------------------------------------------------------------------------
+ WARNING
+
+    By specifying "serviceType=LoadBalancer" you have most likely exposed the
+    Elasticsearch service externally. Please note that Elasticsearch does not
+    implement a authentication mechanism to secure your cluster.
```
Comment (prydonius) [inline incubator/elasticsearch/templates/NOTES.txt]: +1

File: incubator/elasticsearch/templates/NOTES.txt [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    For security reasons, we strongly suggest that you switch to using "ClusterIP"
+    or "NodePort" as the "serviceType".
+-------------------------------------------------------------------------------
+{{- end }}
+
+** Please be patient while the chart is being deployed **
+
+  Elasticsearch can be accessed within the cluster on port {{ .Values.elasticsearchPortNumber }} at {{ template "fullname" . }}.{{ .Release.Namespace }}.svc.cluster.local
+
+  To access for outside the cluster execute the following commands:
```
Comment (prydonius) [inline incubator/elasticsearch/templates/NOTES.txt]: `for` -> `from`

File: incubator/elasticsearch/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+spec:
+  template:
+    metadata:
+      labels:
+        app: "{{ template "name" . }}"
+        release: "{{ .Release.Name }}"
+      annotations:
+        pod.beta.kubernetes.io/init-containers: '[
```
Comment (prydonius) [inline incubator/elasticsearch/templates/deployment.yaml]: the past two version (1.6, 1.7) support the `initContainers` key in the PodSpec, so I think we can use that, unless you think we need to support 1.5 and less?

Comment (sameersbn) [inline incubator/elasticsearch/templates/deployment.yaml]: updated!

**Pull request merged**